File: sentimental-credit/credit.py
# TODO

def main():
    while (True):
        try:
            n = int(input("Number: "))
        except:
            print("Must be a credit card number")
            continue
        break

    CheckSum(n)


def CheckSum(number):
    n = number
    n2 = n
    n3 = n
    digits1 = 0
    digits2 = 0
    doubled = 0
    first_digit = 0
    second_digit = 0
    sum1 = 0
    sum2 = 0
    sum = 0
    counter = 0
    while n2 != 0:
        digits1 = int((n2/10) % 10)
        digits2 = int(n2 % 10)
        doubled = digits1 * 2
        if doubled >= 10:
            first_digit = int(doubled / 10)
            second_digit = int(doubled % 10)
            doubled = first_digit + second_digit

        sum1 = sum1 + doubled
        sum2 = sum2 + digits2
        n2 = int(n2 / 100)
    sum = sum1 + sum2
    if sum%10 != 0:
        print ("INVALID")
    while (n3 != 0):
        n3 = int(n3 / 10)
        counter += 1

    if counter == 15 and (int(n/(10**(counter-2))) == 37 or int(n/(10**(counter-2))) == 34):
        print("AMEX")
    elif counter == 16 and (int(n/(10**(counter-2))) == 51 or int(n/(10**(counter-2))) == 52 or int(n/(10**(counter-2))) == 53 or int(n/(10**(counter-2))) == 54 or int(n/(10**(counter-2))) == 55):
        print("MASTERCARD")
    elif (counter == 13 or counter == 16) and int(n/(10**(counter-1))) == 4:
        print("VISA")
main()

File: credit/credit.c
#include<stdio.h>
#include<cs50.h>

void CheckSum(long n);

int main(void)
{
    long cc = get_long("Number: ");
    CheckSum(cc);

}

void CheckSum(long n)
{
    long copy = n;
    long copy2= copy;
    long x = 10;
    long y = 1;
    int sum1 = 0;
    int sum2 = 0;
    int sum = 0;
    int holder = 0;
    int holder2= 0;
    int mlt2 = 0;
    int check;
    long n1 = n;
    long n2 = n;
    int len = 0;
    long n3 = n;
    while (copy2 != 0)
    {
        copy2 = copy2/100;

        holder = copy/x % 10;
        holder2 = copy/y % 10;

        x = x * 100;
        y = y * 100;

        mlt2 = holder * 2;

        if (mlt2 >= 10)
        {
            sum1 += (mlt2/10%10) + (mlt2%10);
        }
        else
        {
            sum1 += mlt2;
        }
        sum2 += holder2;

        sum = sum1 + sum2;
    }
    check = sum % 10;

    while(true)
    {
        len++;
        n3 = n3 / 10;
        if (n3 == 0)
            break;
    }

    while (n1 >= 100)
        n1 /= 10;
    while (n2 >= 10)
        n2 /= 10;


    if (check == 0)
        {
            if (len == 15 && ( n1 == 34 || n1 == 37))
            {
                printf("AMEX\n");
            }
            else if (len == 16 && (n1 == 51 || n1 == 52 || n1 == 53 || n1 == 54 || n1 == 55))
            {
                printf("MASTERCARD\n");
            }
            else if ((len == 16 || len == 13) && n2 == 4)
            {
                printf("VISA\n");
            }
            else
                printf ("INVALID\n");

        }
    else
        printf ("INVALID\n");

}

